File: 6_Encryption/solution.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the 'encryption' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

string encryption(string s) {    
    double root = sqrt(s.size());
    int column = ceil(root);
    int row = ceil(s.size()/double(column));
    string res;
    
    for (int i=0;i<column;i++) {
        for (int j=0;j<row;j++) {
            if(i+column*j < s.size()) {
                res += s[i+column*j];
            }
        }
        res+=" ";
    }
    return res;
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = encryption(s);

    fout << result << "\n";

    fout.close();

    return 0;
}

File: 6_Encryption/solution.py
#!/bin/python3
/* azusaw */

import math
import os
import random
import re
import sys

#
# Complete the 'encryption' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#

def encryption(s):
    l = len(s)
    n = int(math.sqrt(l))
    r = n if n * (n + 1) > l else n + 1
    c = n + 1 if n ** 2 != l else n
    res = ""
    for i in range(0, c):
        for j in range(0, r):
            if (i + j * c) < l:
                res += s[i + j * c]
        res += " "
    return res

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = encryption(s)

    fptr.write(result + '\n')

    fptr.close()
